.github/workflows/codecov.yml

```diff
@@ -30,27 +30,19 @@ jobs:
           cpu: amd64
           shell: bash
 
-      - name: Restore deps from cache
-        id: deps-cache
-        uses: actions/cache@v3
-        with:
-          path: nimbledeps
-          key: nimbledeps-${{ hashFiles('.pinned') }}
-
       - name: Install deps
-        if: ${{ steps.deps-cache.outputs.cache-hit != 'true' }}
         run: |
-          nimble install_pinned
+          nimble -y install --lock-file=nimble-version-1-6.lock
 
       - name: Run
         run: |
           sudo apt-get update
           sudo apt-get install -y lcov build-essential git curl
           mkdir coverage
           export NIMFLAGS="--lineDir:on --passC:-fprofile-arcs --passC:-ftest-coverage --passL:-fprofile-arcs --passL:-ftest-coverage"
-          nimble testnative
-          nimble testpubsub
-          nimble testfilter
+          nimble testnative --lock-file=nimble-version-1-6.lock
+          nimble testpubsub --lock-file=nimble-version-1-6.lock
+          nimble testfilter --lock-file=nimble-version-1-6.lock
```

Comment:
Does that mean that everytime we want to run tests we'll have to do
```
nimble testnative --lock-file=nimble-version-1-6.lock
# instead of just
nimble testnative
```
?

Reply:
You should be able to do `nimble setup --lock-file=nimble-version-1-6.lock` and from now on it won't need `--lock-file=nimble-version-1-6.lock` . `nimble setup` call will create `nimble.paths` and they will be used when there is no lock file specified.

Reply:
Maybe it's cleaner to do nimble setup in the CI too instead of passing the lock file everytime?

Reply:
this is what I was doing initially, but @zah opposed based on the argument that `setup` is something that only the user should do for local development.

Reply:
Ok, don't care particularly about it, but is there a specific reason it's only for users?

Reply:
I am not 100% sure, I guess it is related to polluting the workspace with nimble.paths file. I will let @zah elaborate.

Reply:
It's a matter of hygiene. If some of the Nimble commands don't work when `nimble setup` hasn't been executed, this means that we've left a bug somewhere in Nimble.

The primary goal of `nimble setup` is to enable the local use of the `nim` executable (and other tools such as `nimsuggest`) without wrapping their executions in a `nimble` command that setups the necessary paths.